src/openrct2/ride/Ride.cpp

```diff
@@ -1753,35 +1761,79 @@ Staff* ride_get_assigned_mechanic(Ride* ride)
 
 /**
  *
- *  rct2: 0x006ABE85
+ *  Circus music is a sound object, rather than music. Needs separate processing.
  */
-static void ride_music_update(Ride* ride)
+static void circus_music_update(Ride* ride)
 {
-    // The circus does not have music in the normal sense - its “music” is a sound effect.
-    if (ride->type == RIDE_TYPE_CIRCUS)
+    Vehicle* vehicle = GetEntity<Vehicle>(ride->vehicles[0]);
+    if (vehicle == nullptr || vehicle->status != Vehicle::Status::DoingCircusShow)
     {
-        Vehicle* vehicle = GetEntity<Vehicle>(ride->vehicles[0]);
-        if (vehicle == nullptr || vehicle->status != Vehicle::Status::DoingCircusShow)
-        {
-            ride->music_tune_id = 255;
-            return;
-        }
+        ride->music_position = 0;
+        ride->music_tune_id = 255;
+        return;
     }
-    else
+
+    // Oscillate parameters for a power cut effect when breaking down
```

Comment:
From here down, it looks like the code is exactly the same as `ride_music_update()`. As such, it should be put into a function, rather than be duplicated.